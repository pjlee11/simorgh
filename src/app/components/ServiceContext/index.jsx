import React from 'react';
import { func, node, string } from 'prop-types';
import services from '../../lib/config/services';

/* Create ServiceContext using the default service */
export const ServiceContext = React.createContext(services.default);

export const ServiceContextProvider = ({ children, service }) => (
  <ServiceContext.Provider value={services[service]}>
    {children}
  </ServiceContext.Provider>
);

export const ServiceContextConsumer = ({ children }) => (
  <ServiceContext.Consumer>{children}</ServiceContext.Consumer>
);

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
};

ServiceContextConsumer.propTypes = {
  children: func.isRequired,
};
