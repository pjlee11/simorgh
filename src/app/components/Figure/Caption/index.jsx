import React from 'react';
import { node } from 'prop-types';
import styled from 'styled-components';
import {
  FF_NEWS_SANS_REG,
  C_STORM,
  GEL_SPACING,
  GEL_SPACING_DBL,
  C_STONE,
} from '../../../lib/constants/styles';
import VisuallyHiddenText from '../../VisuallyHiddenText';
import mediaQuery from '../../../helpers/mediaQueries';
import { T_LONG_PRIMER } from '../../../lib/constants/typography';
import { ServiceContext } from '../../ServiceContext';

const StyledCaption = styled.figcaption`
  ${T_LONG_PRIMER};
  background-color: ${C_STONE};
  color: ${C_STORM};
  font-family: ${FF_NEWS_SANS_REG};
  padding: ${GEL_SPACING};
  width: 100%;
  ${mediaQuery.laptopAndLarger} {
    padding: ${GEL_SPACING} ${GEL_SPACING_DBL};
  }
`;

const Caption = ({ children }) => (
  <StyledCaption>
    <ServiceContext.Consumer>
      {({ imageCaptionOffscreenText }) => (
        <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
      )}
    </ServiceContext.Consumer>
    {children}
  </StyledCaption>
);

Caption.propTypes = {
  // children will be "element.isRequired" in the future to support embedded <lang> and markdown
  children: node.isRequired,
};

export default Caption;
