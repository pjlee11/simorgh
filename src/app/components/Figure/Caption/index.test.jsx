import React from 'react';
import { shouldMatchSnapshot } from '../../../helpers/tests/testHelpers';
import { ServiceContext } from '../../ServiceContext';
import Caption from './index';

const CaptionWithContext = (captionText, contextStub) => (
  <ServiceContext.Provider value={contextStub}>
    <Caption>{captionText}</Caption>
  </ServiceContext.Provider>
);

const newsServiceContextStub = {
  imageCaptionOffscreenText: 'Image caption, ',
};
const persianServiceContextStub = {
  imageCaptionOffscreenText: ' ، عنوان تصویر',
};

shouldMatchSnapshot(
  'should render Caption with news service context',
  CaptionWithContext('This is some Caption text', newsServiceContextStub),
);

shouldMatchSnapshot(
  'should render Caption with persian service context',
  CaptionWithContext('توصیف چیزی که اتفاق می افتد', persianServiceContextStub),
);
