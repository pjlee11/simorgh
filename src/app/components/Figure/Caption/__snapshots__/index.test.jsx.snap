// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Caption with news service context 1`] = `
.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  top: 0;
}

.c0 {
  font-size: 0.9375em;
  line-height: 1.125rem;
  background-color: #D5D0CD;
  color: #404040;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  padding: 0.5rem;
  width: 100%;
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 0.875em;
  }
}

@media (min-width:37.5em) {
  .c0 {
    padding: 0.5rem 1rem;
  }
}

<figcaption
  className="c0"
>
  <span
    className="c1"
  >
    Image caption, 
  </span>
  This is some Caption text
</figcaption>
`;

exports[`should render Caption with persian service context 1`] = `
.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  top: 0;
}

.c0 {
  font-size: 0.9375em;
  line-height: 1.125rem;
  background-color: #D5D0CD;
  color: #404040;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  padding: 0.5rem;
  width: 100%;
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 0.875em;
  }
}

@media (min-width:37.5em) {
  .c0 {
    padding: 0.5rem 1rem;
  }
}

<figcaption
  className="c0"
>
  <span
    className="c1"
  >
     ، عنوان تصویر
  </span>
  توصیف چیزی که اتفاق می افتد
</figcaption>
`;
